chainerx_cc/chainerx/cuda/cuda_device/arithmetic.cu

```diff
@@ -68,28 +46,7 @@ public:
 
 CHAINERX_CUDA_REGISTER_OP(AddASOp, CudaAddASOp);
 
-template <typename T>
-struct SubtractImpl {
-    using CudaType = cuda_internal::DataType<T>;
-    __device__ void operator()(int64_t /*i*/, CudaType x1, CudaType x2, CudaType& out) { out = ArithmeticOps<CudaType>::Subtract(x1, x2); }
-};
-
-class CudaSubtractOp : public SubtractOp {
-public:
-    void Call(const Array& x1, const Array& x2, const Array& out) override {
-        Device& device = x1.device();
-        device.CheckDevicesCompatible(x1, x2, out);
-        const Array& x1_cast = x1.dtype() == out.dtype() ? x1 : x1.AsType(out.dtype());
-        const Array& x2_cast = x2.dtype() == out.dtype() ? x2 : x2.AsType(out.dtype());
-        CudaSetDeviceScope scope{device.index()};
-        VisitNumericDtype(out.dtype(), [&](auto pt) {
-            using T = typename decltype(pt)::type;
-            Elementwise<const T, const T, T>(SubtractImpl<T>{}, x1_cast, x2_cast, out);
-        });
-    }
-};
-
-CHAINERX_CUDA_REGISTER_OP(SubtractOp, CudaSubtractOp);
+CHAINERX_CUDA_REGISTER_ELTWISE_DTYPE_BINARY_OP(Subtract, { out = ArithmeticOps<CudaType>::Subtract(x1, x2); }, VisitNumericDtype);
```

Comment:
I'm wondering if we should use the macro as follows, requiring the whole class name including the "Op" suffix. Benefits are:
- More consistent with `CHAINERX_{NATIVE,CUDA}_REGISTER_OP`.
- Better grep-ability.
- Appending "Op" is convention rather than an enforcement.

What do you think?
```suggestion
CHAINERX_CUDA_REGISTER_ELTWISE_DTYPE_BINARY_OP(SubtractOp, { out = ArithmeticOps<CudaType>::Subtract(x1, x2); }, VisitNumericDtype);
```

Reply:
Makes sense. Ties up with the original `Op` structure as well.